scripts/helm-fan-out.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+# Based on https://github.com/helm/helm/issues/4680#issuecomment-1152289297
+# helm-fan-out
+
+if [ -z "$1" ]; then
+    echo "Please provide an output directory"
+    exit 1
+fi
+
+awk -vout="$1" -F": " '
+  $0~/^# Source: / {
+    file=out"/"$2;
+    if (!(file in filemap)) {
+      filemap[file] = 1
+      print "Creating "file;
+      system ("mkdir -p $(dirname "file")");
+    }
+    print "---" >> file;
+  }
+  $0!~/^# Source: / {
+    if ($0!~/^---$/) {
+      if (file) {
+        print $0 >> file;
+      }
+    }
+  }'
```

Comment:
I think we can use `helm template` to render the individual files by adopting the `--show-only` flag. As in the following example:

```sh
for t in $(ls -1 templates/task-*.yaml); do
    task_file="$(basename ${t})"
    echo "# Rendering template ${t} into ${task_file}"
    make helm-template ARGS="--show-only=${t}" >/tmp/${task_file}
done
```

A better place for this is the `Makefile`, adding a new `helm-template-tasks` target, i.e.:


```Makefile
helm-template-tasks:
    for t in $(ls -1 templates/task-*.yaml); do
        helm template --show-only=${t} >/tmp/`basename ${t}``
    done
```

With a configurable directory instead of `/tmp` like on the example above. So the task repository can prepare it's own files for later release.

WDYT?

Reply:
Interesting, I didn't know that, it would definitely be better then.